core/execution/equity_shares.go

```diff
@@ -161,27 +158,40 @@ func (es *EquityShares) SetPartyStake(id string, newStakeU *num.Uint) {
 		v.stake = newStake
 		es.totalVStake = es.totalVStake.Add(v.vStake)
 		es.totalPStake = es.totalPStake.Add(v.stake)
-		v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
+		// average entry valuation should not be changed when reducing the physical stake
+		// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
```

Comment:
Is that meand to be here?

Reply:
Not anymore, I'll remove most stuff I commented out